### DISCUSSION TITLE
Cannot set permission android

### DISCUSSION THREAD

**Discussion by hyperyuiz** [airsdk/Adobe-Runtime-Support#3068] [Q&A] [answered]
AIR for Android: 50.2.4.3

I cannot install the following permissions:
```
<uses-permission android:name="com.google.android.gms.permission.AD_ID"/>
<uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE" />
```  

When I built the apk version for my application and uploaded it in google console, a warning appeared `com.google.android.gms.permission.AD_ID has not been permission `.    

I tried many ways like below but it doesn't work:

```
<android>
	<manifestAdditions><![CDATA[
		<manifest android:installLocation="auto">
				<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
				<uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
				<uses-permission android:name="android.permission.READ_PHONE_STATE"/>
				<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
				<uses-permission android:name="com.google.android.gms.permission.AD_ID"/>
				<uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE" />
		</manifest>
	]]></manifestAdditions>
</android>
```

Please help me!

**Comment by marchbold** [comment]
Can you elaborate on where the error occurred and the exact message?

Adding these shouldn't affect the build however it can affect the responses you need to provide for the data privacy survey in the play console.

**Comment by hyperyuiz** [comment]
@marchbold 
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/29688661/17083b3d-0110-4512-af30-a7bbb5cab9d4)  

when I try to add ```<uses-permission android:name="com.google.android.gms.permission.AD_ID"/>```  it seems to fail on build

**Comment by marchbold** [comment]
That error is saying it can't find a resource `google_play_services_version`, are you sure you haven't added additional items to your manifest and forgotten an ANE dependency?

**Comment by hyperyuiz** [comment]
@marchbold   
Ah, I got it.   
Sorry I'm a newbie.  
Can you guide me to add it 🙇‍♂️

**Comment by marchbold** [comment, marked as answer]
I'm going to assume you are using our standard extensions, so this resource is contained in [`com.distriqt.playservices.Base`](https://github.com/distriqt/ANE-GooglePlayServices/blob/master/lib/com.distriqt.playservices.Base.ane)

You'll need to download and add that extension to your build (depends on IDE but can read guides here: https://docs.airnativeextensions.com/docs/tutorials/getting-started) and add the extension id to the `extensions` list in your app descriptor:

```xml
<extensions>
    <extensionID>com.distriqt.playservices.Base</extensionID>
</extensions>
```

Highly suggest you go back and check the documentation for whatever extension you are using though as this should have been covered there :) 

Alternatively if you are using `apm`:

```
apm install com.distriqt.playservices.Base
apm generate app-descriptor
```

**Comment by hyperyuiz** [comment]
@marchbold   
Thanks for your help!  
I will learn about it ( I just want to show ads in my app)  

I found an extension that can help me do that
https://airnativeextensions.com/extension/com.distriqt.Adverts
